pennylane/boolean_fn.py

```diff
@@ -82,13 +83,16 @@ def __init__(self, fn):
         functools.update_wrapper(self, fn)
 
     def __and__(self, other):
-        return BooleanFn(lambda obj: self.fn(obj) and other.fn(obj))
+        return BooleanFn(lambda *args, **kwargs: self(*args, **kwargs) and other(*args, **kwargs))
 
     def __or__(self, other):
-        return BooleanFn(lambda obj: self.fn(obj) or other.fn(obj))
+        return BooleanFn(lambda *args, **kwargs: self(*args, **kwargs) or other(*args, **kwargs))
 
     def __invert__(self):
-        return BooleanFn(lambda obj: not self.fn(obj))
+        return BooleanFn(lambda *args, **kwargs: not self(*args, **kwargs))
 
-    def __call__(self, obj):
-        return self.fn(obj)
+    def __call__(self, *args, **kwargs):
+        return self.fn(*args, **kwargs)
+
+    def __repr__(self):
+        return f"BooleanFn({self.fn.__name__})"
```

Comment:
:thinking: what if a user has combined two boolean functions with an `&`, I guess the repr here would be for a lambda function?

Reply:
True, I had fixed this for my `And` objects, I can try to do something similar here.